packages/http-server/src/server.ts

```diff
@@ -60,72 +63,78 @@ export const createServer = (operations: IHttpOperation[], opts: IPrismHttpServe
     };
 
     request.log.info({ input }, 'Request received');
-    try {
-      const operationSpecificConfig = getHttpConfigFromRequest(input);
-      const mockConfig = opts.config.mock === false ? false : { ...opts.config.mock, ...operationSpecificConfig };
 
-      const response = await prism.request(input, operations, {
-        ...opts.config,
-        mock: mockConfig,
-      });
+    const operationSpecificConfig = getHttpConfigFromRequest(input);
+    const mockConfig = opts.config.mock === false ? false : { ...opts.config.mock, ...operationSpecificConfig };
 
-      const { output } = response;
+    return pipe(
+      prism.request(input, operations, { ...opts.config, mock: mockConfig }),
+      TaskEither.chain(response => {
+        const { output } = response;
 
-      const inputOutputValidationErrors = response.validations.output
-        .map(createErrorObjectWithPrefix('response'))
-        .concat(response.validations.input.map(createErrorObjectWithPrefix('request')));
+        const inputOutputValidationErrors = response.validations.output
+          .map(createErrorObjectWithPrefix('response'))
+          .concat(response.validations.input.map(createErrorObjectWithPrefix('request')));
 
-      reply.header('sl-violations', JSON.stringify(inputOutputValidationErrors));
+        if (inputOutputValidationErrors.length > 0) {
+          reply.header('sl-violations', JSON.stringify(inputOutputValidationErrors));
 
-      if (inputOutputValidationErrors.length > 0) {
-        const errorViolations = inputOutputValidationErrors.filter(
-          v => v.severity === DiagnosticSeverity[DiagnosticSeverity.Error]
-        );
-        if (opts.errors && errorViolations.length > 0) {
-          throw ProblemJsonError.fromTemplate(
-            VIOLATIONS,
-            'Your request/response is not valid and the --errors flag is set, so Prism is generating this error for you.',
-            { validation: errorViolations }
+          const errorViolations = inputOutputValidationErrors.filter(
+            v => v.severity === DiagnosticSeverity[DiagnosticSeverity.Error]
           );
+          if (opts.errors && errorViolations.length > 0) {
```

Comment:
I believe this code still needs some love and we will do it later?

Reply:
Yes, this was the minimal part to move it over. We'll keep iterating on this.